### ISSUE TITLE
[Bug]: Input of type "number" and inputMode "decimal" doesn't pass a value if a non-digit or second demical are entered

### ISSUE THREAD

**Issue by jkirsch-LF** [cloudscape-design/components#1947]
### Browser

Firefox

### Package version

v3.0.500

### React version

v18.2.0

### Description

For the `Input` component, if this configuration is used:

```tsx
<Input
    onChange={({ detail }) => {
        console.log("DETAIL: ", detail);
        setValue(detail.value);
    }}
    value={value}
    inputMode="decimal"
    step={0.1}
    type="number"
/>
```

The value of `detail.value` is `""` if a non-digit or second decimal input are entered.
The behavior around the input `0.n` is also very inconsistent.  If the current value is 0.1 and the user presses backspace, the `value` is `""`, rather than `0.` or even `0.0`. 

The main issue here is that trying to apply constraints on the value becomes unnecessarily complex if the value passed is empty or incorrect.  

I noticed this when trying to apply a function to the `detail.value` during the `onChange` method that would prevent the user from inputting an incorrectly formatted number (0.01.0, 24ab, etc).  For something like an input of a letter, I can deduce from the `detail.value` that the input was incorrect (ie: `1478j`).  That same behavior doesn't work when trying to update a value from `0.3` to `0.2`, or something like that. In this example, when I delete the `3` from `0.3`, I get an empty `value`, when the value should be `0.`.

In addition, the actual value being displayed still allows any character to be displayed, but the `detail.value` will continue to be `""` until the number becomes valid.  Continuing to edit the value while malformed will also eventually cause the `onEvent` method not to fire.  In the reproduction link below, if you open the console, and in the first input enter another decimal (or any non-digit) and then start pressing any key you'll notice the console.logs stop printing after 5ish inputs.

tldr: The value of `event.detail.value` from `onChange` is incorrect when the input creates a malformed number, but the malformed number still displays inside the input.

### Reproduction

[Reproduction Link](https://codesandbox.io/p/sandbox/funny-currying-d8jcvq?file=%2Fsrc%2FApp.js%3A1%2C1-79%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsdnyqrw00062e65axalehg5%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsdnyqrw00022e65tqw3xxhr%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsdnyqrw00032e65jzp2dcux%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsdnyqrw00052e650s7uvpac%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsdnyqrw00022e65tqw3xxhr%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsdnyqrw00012e65xpj8qm4n%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clsdo0c58005g2e65ua029cda%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A1%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A79%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clsdnyqrw00022e65tqw3xxhr%2522%252C%2522activeTabId%2522%253A%2522clsdo0c58005g2e65ua029cda%2522%257D%252C%2522clsdnyqrw00052e650s7uvpac%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsdnyqrw00042e65ua5ij7di%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clsdnyqrw00052e650s7uvpac%2522%252C%2522activeTabId%2522%253A%2522clsdnyqrw00042e65ua5ij7di%2522%257D%252C%2522clsdnyqrw00032e65jzp2dcux%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clsdnyqrw00032e65jzp2dcux%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/cloudscape-design/components/blob/main/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/cloudscape-design/components/issues) for duplicate problems

**Comment by just-boris** [comment]
The behavior you report is the native behavior of the plain `<input type="number">`.  We cannot fix this for you, because this is all information available from the web platform.

The typical way to deal with this issues is to apply validation only on the input blur, when the user input is definitely complete.

If you see any inconsistency between our component and the plain element, we can investigate this as a bug report.

**Comment by just-boris** [comment]
Closing due to inactivity. Feel free to reopen if you have further questions
